gapic-generator/lib/gapic/schema/wrappers.rb

```diff
@@ -780,11 +783,11 @@ class Resource
       def initialize descriptor
         @parent = nil
         @descriptor = descriptor
-        @parsed_patterns = descriptor.pattern.map do |pattern|
-          pattern.split("/").map do |segment|
-            segment =~ %r{\{[^/\}]+(=[^\}]+)?\}} ? "*" : segment
-          end.freeze
+        patterns = descriptor.pattern.map do |pattern|
+          Gapic::PathPattern.parse pattern
         end.freeze
+        @parsed_patterns = patterns.map(&:template).compact.uniq.freeze
+        @parsed_parent_patterns = patterns.map(&:parent_template).compact.uniq.freeze
```

Comment:
`Array#compact` filters out `nil`s only, not empty strings, whereas your `Gapic::PathPattern::Pattern#parent_template` appears to return empty string if there's no possible parent. Either have to return `nil` from `parent_template` in that case, or alter the logic here. And better put a test case around this. :)

Reply:
definitely. I've added a unit and an integration test.